lib/BloqadeExpr/src/lower.jl

```diff
@@ -202,6 +204,30 @@ function YaoBlocks.Optimise.to_basictypes(ex::RydInteract)
     return term
 end
 
+function YaoBlocks.Optimise.to_basictypes(h::RydbergHamiltonian{Nothing,Nothing})
```

Comment:
these `to_basictypes` should be just one overload with `if else`s, so no dynamic dispatch will happen. If we are gonna define `add_terms` maybe should directly use `add_terms` here?